Mathlib/Analysis/Normed/Ring/Lemmas.lean

```diff
@@ -214,3 +215,60 @@ instance Int.instNormOneClass : NormOneClass ℤ :=
 
 instance Int.instNormMulClass : NormMulClass ℤ :=
   ⟨fun a b ↦ by simp [← Int.norm_cast_real, abs_mul]⟩
+
+section NormMulClass
+
+variable [NormedRing α] [NormMulClass α] {a : α}
+
+lemma antilipschitzWith_mul_left {a : α} (ha : a ≠ 0) : AntilipschitzWith (‖a‖₊⁻¹) (a * ·) :=
+  AntilipschitzWith.of_le_mul_dist fun _ _ ↦ by simp [dist_eq_norm, ← _root_.mul_sub, ha]
+
+lemma antilipschitzWith_mul_right {a : α} (ha : a ≠ 0) : AntilipschitzWith (‖a‖₊⁻¹) (· * a) :=
+  AntilipschitzWith.of_le_mul_dist fun _ _ ↦ by
+    simp [dist_eq_norm, ← _root_.sub_mul, ← mul_comm (‖a‖), ha]
+
+/-- Multiplication by a nonzero element `a` on the left, as a `Dilation` of a ring with a strictly
+multiplicative norm. -/
+@[simps!]
+def Dilation.mulLeft {a : α} (ha : a ≠ 0) : α →ᵈ α where
```

Comment:
Out of scope for this PR, but perhaps we should change this in a later pr to `ha : ‖a‖ ≠ 0` to make this work for `SeminormedRing`.

Reply:
That should indeed be possible, but it's not 100% clear to me if the gain in generality is worth the loss of convenience in the main use-case of actually normed rings. Let's save it for later.